File: backend/app/schemas/rfi.py
from __future__ import annotations

from pydantic import BaseModel
from datetime import datetime, date


class RFICreate(BaseModel):
    rfi_number: str | None = None
    title: str
    question: str
    status: str = "Open"


class RFIOut(BaseModel):
    id: str
    project_id: str
    rfi_number: str | None
    title: str
    question: str
    status: str
    created_at: datetime
    due_date: date | None
    answered_at: datetime | None

    phase_classification: str | None
    issue_classification: str | None
    entities: dict | None

    trade_name: str | None
    csi_division: str | None
    drawing_reference: str | None
    gridline_reference: str | None
    room_number: str | None
    spec_section: str | None
    responsible_party: str | None
    cost_impact: float | None
    schedule_impact_days: int | None

    class Config:
        from_attributes = True

File: frontend/src/api/rfi.ts
import { api } from './client'

export type RFI = {
  id: string
  rfi_number?: string
  title: string
  question: string
  status: string
  created_at: string
  phase_classification?: string
  issue_classification?: string
  trade_name?: string
  drawing_reference?: string
  spec_section?: string
  gridline_reference?: string
  room_number?: string
  cost_impact?: number
  schedule_impact_days?: number
}

export async function listRfis(projectId: string): Promise<RFI[]> {
  const res = await api.get(`/projects/${projectId}/rfis`)
  return res.data
}

export async function createRfi(projectId: string, payload: { rfi_number?: string; title: string; question: string; status?: string }): Promise<RFI> {
  const res = await api.post(`/projects/${projectId}/rfis`, payload)
  return res.data
}

export async function analyzeRfi(projectId: string, rfiId: string) {
  const res = await api.post(`/projects/${projectId}/rfis/${rfiId}/analyze`)
  return res.data
}

export async function importRfisCsv(projectId: string, file: File) {
  const form = new FormData()
  form.append('file', file)
  const res = await api.post(`/projects/${projectId}/rfis/import_csv`, form, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
  return res.data
}
